From e7d3f651c7fe39ce518bc83b0246f3dd5fd8eed8 Mon Sep 17 00:00:00 2001
From: Jeremie GARCIA <jgarci4X>
Date: Fri, 28 Sep 2012 18:35:08 +0200
Subject: [PORT FROM MAIN] wifi: Add improved interface logging from framework to wpa_supplicant

BZ: 59886

[DO NOT UPSTREAM]
When upgrading to Jelly Bean, we've lost all wifi HAL command prints.
This is due to a missing patch port:
   http://android.intel.com:8080/#/c/36209/

However this patch has a 1024 char maximum print limitation.

So this patch ports most of original one with little improvement.

Feature   : Wlan
Category  : Improvement
Purpose   : AOSP+
Origin    : Upstream
Upstream-candidate: no

Change-Id: I939789369e0bf0ac7e94f36e4b22704f56c639d6
Orig-Change-Id: I65b52dd8dd65c997a32a79a206d514145ff18565
Signed-off-by: Jeremie GARCIA <jeremiex.garcia@intel.com>
---
 wifi/wifi.c |   30 +++++++++++++++++++++++++++++-
 1 files changed, 29 insertions(+), 1 deletions(-)

diff --git a/wifi/wifi.c b/wifi/wifi.c
index 12b5966..4fe4920 100644
--- a/wifi/wifi.c
+++ b/wifi/wifi.c
@@ -707,6 +707,31 @@ int wifi_connect_to_supplicant(const char *ifname)
     }
 }
 
+void log_cmd(const char *cmd)
+{
+    LOGI("CMD: %s\n", cmd);
+}
+
+void log_reply(char *reply, size_t *reply_len)
+{
+    char replyLocal[*reply_len];
+    char delims[] = "\n";
+    char *result = NULL;
+
+    strncpy(replyLocal, reply, *reply_len);
+
+    if (*reply_len > 0 && replyLocal[*reply_len-1] == '\n')
+        replyLocal[*reply_len-1] = '\0';
+    else
+        replyLocal[*reply_len] = '\0';
+
+    result = strtok(replyLocal , delims );
+    while( result != NULL ) {
+        LOGI("REPLY: %s\n", result);
+        result = strtok( NULL, delims );
+    }
+}
+
 int wifi_send_command(int index, const char *cmd, char *reply, size_t *reply_len)
 {
     int ret;
@@ -715,6 +740,7 @@ int wifi_send_command(int index, const char *cmd, char *reply, size_t *reply_len
         ALOGV("Not connected to wpa_supplicant - \"%s\" command dropped.\n", cmd);
         return -1;
     }
+    log_cmd(cmd);
     ret = wpa_ctrl_request(ctrl_conn[index], cmd, strlen(cmd), reply, reply_len, NULL);
     if (ret == -2) {
         ALOGD("'%s' command timed out.\n", cmd);
@@ -722,11 +748,13 @@ int wifi_send_command(int index, const char *cmd, char *reply, size_t *reply_len
         TEMP_FAILURE_RETRY(write(exit_sockets[index][0], "T", 1));
         return -2;
     } else if (ret < 0 || strncmp(reply, "FAIL", 4) == 0) {
+        LOGI("REPLY: FAIL\n");
         return -1;
     }
     if (strncmp(cmd, "PING", 4) == 0) {
         reply[*reply_len] = '\0';
     }
+    log_reply(reply, reply_len);
     return 0;
 }
 
@@ -812,7 +840,7 @@ int wifi_wait_on_socket(int index, char *buf, size_t buflen)
             memmove(buf, match+1, nread+1);
         }
     }
-
+    LOGI("EVENT: %s\n", buf);
     return nread;
 }
 
-- 
1.7.4.1

